fix: correct reading of sync-labels input.

adam-azarchs opened the pull request:
**Description:**
Correctly check the truth value of the `sync-labels` input.
Contrary to the assumptions made in the unit tests, core.getInput always returns a string, and `!!'false'` is true.

**Related issue:**
Fixes https://github.com/actions/labeler/issues/112

**Check list:**
- [x] Mark if documentation changes are required.
- [x] Mark if tests were added or updated to cover the changes.

BREAKING CHANGE: previously, any non-empty string would result in labels being deleted.  With this change, an empty string will cause an error, but strings representing yaml-compliant boolean values will be correctly interpreted.

adam-azarchs commented:
(granted it's _still_ a breaking change, but only for people who were setting `sync-labels` to something other than a true/false/empty string, and the behavior for those people was probably not what they'd expected anyway)

MaksimZhukov commented:
@adam-azarchs thank you for the explanation!
As you mentioned, your PR introduces a breaking change either way, so we will release a new major version of the action. Everyone who uses the current version will not be broken, so there is no need for backward compatibility.

adam-azarchs commented:
Thinking more about it, I agree, since the failure mode will be the action failing loudly, as opposed to just not doing what the user wanted.  I'll make that change.

MaksimZhukov reviewed (commented):
@adam-azarchs thank you for resolving the comment, I left a few more, please take a look.

adam-azarchs commented on __tests__/main.test.ts:
```diff
@@ -50,13 +50,18 @@ describe('run', () => {
     jest
       .spyOn(core, 'getInput')
       .mockImplementation((name: string, ...opts) => mockInput[name]);
+    jest
+      .spyOn(core, 'getBooleanInput')
+      .mockImplementation((name: string, opts) => {
+        if (name !== 'sync-labels') throw new Error(`invalid input ${name}`);
+        return true;
+      });
```
Unfortunately that doesn't work, because `core.getBooleanInput` doesn't call the mocked version of `core.getInput`.

MaksimZhukov replied:
I guess the following code snippet should work fine:
```ts
jest
  .spyOn(core, 'getBooleanInput')
  .mockImplementation((name: string, ...opts) => mockInput[name]);
```

adam-azarchs replied:
Done.

adam-azarchs replied:
That said, the other problem with this approach is it would not catch regressions of the same bug.  I can enhance the test a bit for that, however.

adam-azarchs replied:
I've further updated it to make sure that the `mockInput` is a string-string dictionary so that if one calls `getInput` they'll still get a string, and to make `getBooleanInput` return `=== 'true'` (which is perhaps oversimplified but adequate for purposes of a test mock).

adam-azarchs commented:
Thanks for the review so far!   I believe I've addressed all of the feedback at this point; please let me know if there is anything further to do (I'm not in a rush to merge this or anything, but I want to know if I can cross it off my to-do list...)

MaksimZhukov commented:
Hello @adam-azarchs! Thank you!
I will ask my team members to review this PR as well. 
We will merge it when the changes from [this PR](https://github.com/actions/labeler/pull/203) are ready. We would like to include them in one release.

dfandrich commented:
There are 23 other open PRs right now. Are any of them planned to get into the next release as well?

MaksimZhukov commented:
Hello @dfandrich ! 
We are currently thoroughly investigating the issues and pull requests in the repository, so most likely some changes will be also included or released separately

dfandrich commented:
Super!

ghusse commented:
Is there any news regarding the release of this fix? The bug is quite annoying.

MaksimZhukov commented:
Hello @ghusse! We apologize for the inconvenience!
We will merge it when the changes from [this PR](https://github.com/actions/labeler/pull/203) are ready. We would like to include them in one major release.

MaksimZhukov commented:
Hello @adam-azarchs! Could you please resolve the conflicts? We would like to merge the PR
